### PR TITLE
(#214) - don't stop indexing if there's a user error in map/reduce

### PR DESCRIPTION
**Pull request by nolanlawson**
@ddouglascarr This is my recommended fix for #214. It shows this error in the console, and does *not* stop indexing:

screenshot 2015-02-14 11 09 02

See the added tests that verify this is working. (As it turns out, we don't need to remove the listener, because the `db` object gets destroyed.)

### CODE REVIEW

**Comment by nolanlawson** [comment]
Interestingly, this brings us up to parity with CouchDB, which apparently does not stop indexing if there's an error in a certain document. It just treats it as a no-op.

**Comment by nolanlawson** [comment]
Small bug, should be fixed now.

File: index.js [hunk 1]
```diff
@@ -69,12 +82,10 @@ function rowToDocId(row) {
 }
 
 function createBuiltInError(name) {
-  var error = new Error('builtin ' + name +
+  var message = 'builtin ' + name +
     ' function requires map values to be numbers' +
-    ' or number arrays');
-  error.name = 'invalid_value';
-  error.status = 500;
-  return error;
```
Comment (nolanlawson) [inline index.js]: I decided to harmonize this error with the style we were using for our other errors. So I added a new BuiltInError class.

File: test/test.js [hunk 2]
```diff
@@ -1253,7 +1253,7 @@ function tests(dbName, dbType, viewType) {
       });
 
       it('many simultaneous persisted views', function () {
-        this.timeout(30000);
```
Comment (nolanlawson) [inline test/test.js]: This test was taking forever for me in Chrome's IndexedDB.

**Comment by ddouglascarr** [comment]
Thanks for that, it looks good. 

I've closed my PR. 

Ahh, I'm new here, is there anything else I should do?

**Comment by nolanlawson** [comment]
Yes, we always love having new contributors around, especially good ones. :)

There's a list of open issues marked "helpwanted" both [in mapreduce](https://github.com/pouchdb/mapreduce/labels/helpwanted) and [in the main repo](https://github.com/pouchdb/pouchdb/labels/help%20wanted). If you find something that looks interesting or that you'd like to see fixed, then don't hesitate to have at it! :)

**Comment by daleharvey** [comment]
+1 looks good

**Pull request merged**